### PR TITLE
button header fixes

### PR DESCRIPTION
**Pull request by Destwood**

### CODE REVIEW

File: .vscode/settings.json
Comment (MAKSsMaK) [inline .vscode/settings.json:1]: delete it from git and added this file to ignore

File: src/shared/modules/Header/Header.module.scss [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
-    box-shadow: 0px 5px 18px 0px #000;
-}
\ No newline at end of file
+  height: fit-content;
+  display: flex;
+  justify-content: space-between;
+  padding: 0.7rem 1rem;
+  background-color: #171c26;
+  box-shadow: 0px 3px 1px 0px #0000009d;
+}
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.module.scss:8]: add enter

File: src/shared/modules/Header/Header.tsx [hunk 2]
```diff
@@ -1,8 +1,38 @@
...
+import React, { useState } from "react";
 import style from "./Header.module.scss";
+import Button from "../../components/Button";
+import logo from "../../../assets/logo.webp";
+import { Link } from "react-router-dom";
+import Input from "../../components/Input";
 
 function Header() {
-    return ( <div className={style.header}></div> );
+  const [searchValue, setSearchValue] = useState("");
```
Comment (MAKSsMaK) [inline src/shared/modules/Header/Header.tsx]: added typing

File: src/shared/components/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { Theme } from "@mui/material/styles";
+
+interface InputProps {
+  value: string;
+  onChange: (value: string) => void;
+  placeholder?: string;
+  type: "filled" | "outlined";
+}
+
+const commonInputStyles: SxProps<Theme> = {
```
Comment (MAKSsMaK) [inline src/shared/components/Input.tsx]: create a separate file in this directory and take the style there

File: src/shared/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import MuiButton from "@mui/material/Button";
+import { SxProps } from "@mui/system";
+import { Theme } from "@mui/material/styles";
+
+interface ButtonProps {
+  children: string;
+  type: "contained" | "filled" | "outlined";
+}
+
+const commonButtonStyles: SxProps<Theme> = {
```
Comment (MAKSsMaK) [inline src/shared/components/Button.tsx]: create a separate file in this directory and take the style there

File: src/shared/modules/Header/Header.tsx (hunk 2 above)
Comment (Destwood) [reply src/shared/modules/Header/Header.tsx]: протипізувати

File: src/shared/components/Input/InputStyle.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+const commonInputStyles: SxProps<Theme> = {
+  bgcolor: "#2F384C",
+  borderRadius: ".5rem",
+  color: "white",
+  padding: "0",
+  "& .MuiOutlinedInput-notchedOutline": {
+    border: "none",
+  },
+  "& .MuiInputBase-input": {
```
Comment (MAKSsMaK) [inline src/shared/components/Input/InputStyle.tsx:14]: add enter

File: src/shared/components/Input/InputStyle.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { SxProps } from "@mui/system";
+import { Theme } from "@mui/material/styles";
+
+const commonInputStyles: SxProps<Theme> = {
+  bgcolor: "#2F384C",
+  borderRadius: ".5rem",
+  color: "white",
+  padding: "0",
+  "& .MuiOutlinedInput-notchedOutline": {
```
Comment (MAKSsMaK) [inline src/shared/components/Input/InputStyle.tsx:10]: add enter

File: src/shared/components/Input/InputStyle.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    border: "none",
+  },
+  "& .MuiInputBase-input": {
+    color: "white",
+    padding: ".5rem 1rem",
+    fontSize: ".9rem",
+    "&::placeholder": {
```
Comment (MAKSsMaK) [inline src/shared/components/Input/InputStyle.tsx:19]: add enter

File: src/shared/components/Input/InputStyle.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      color: "#fff",
+    },
+  },
+  "&:hover .MuiOutlinedInput-notchedOutline": {
```
Comment (MAKSsMaK) [inline src/shared/components/Input/InputStyle.tsx:24]: add enter

File: src/shared/components/Input/InputStyle.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    border: "1px solid #ffffff2a",
+  },
+  "&:focus-within .MuiOutlinedInput-notchedOutline": {
```
Comment (MAKSsMaK) [inline src/shared/components/Input/InputStyle.tsx:28]: add enter

File: src/shared/components/Button/ButtonStyle.tsx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  border: "1px solid #2f384c",
+  transition: "0.1s",
+  margin: "0 0.5rem",
+};
+
+const buttonStyles: Record<ButtonProps["type"], SxProps<Theme>> = {
+  contained: {
+    ...commonButtonStyles,
+    backgroundColor: "#2f384c",
+    "&:hover": {
```
Comment (MAKSsMaK) [inline src/shared/components/Button/ButtonStyle.tsx:19]: add enter

File: src/shared/components/Button/ButtonStyle.tsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+      backgroundColor: "#4a556d",
+    },
+  },
```
Comment (MAKSsMaK) [inline src/shared/components/Button/ButtonStyle.tsx:22]: add enter

File: src/shared/components/Button/ButtonStyle.tsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  filled: {
+    ...commonButtonStyles,
+    backgroundColor: "#fff",
+    color: "#000",
+    "&:hover": {
```
Comment (MAKSsMaK) [inline src/shared/components/Button/ButtonStyle.tsx:29]: add enter

File: src/shared/components/Button/ButtonStyle.tsx [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+      color: "#fff",
+      border: "1px solid #fff",
+      backgroundColor: "transparent",
+    },
+  },
+  outlined: {
```
Comment (MAKSsMaK) [inline src/shared/components/Button/ButtonStyle.tsx:36]: add enter

File: src/utils/constants/ModalOptions.ts [hunk 14]
```diff
@@ -0,0 +1,6 @@
+
+
```
Comment (MAKSsMaK) [inline src/utils/constants/ModalOptions.ts]: delete enter

**Pull request merged**